### PR TITLE
#860rb60p2 Automate SLAM build to Android

### PR DESCRIPTION
**Pull request by ron-lbb**

### CODE REVIEW

**Comment by guyrodoy1** [comment]
Task linked: [CU-860rb60p2 Integrate SLAM into DemoApp project](https://app.clickup.com/t/860rb60p2)

File: README.md [hunk 1]
```diff
@@ -81,11 +81,26 @@ We provide some examples to process input of a monocular, monocular-inertial, st
...
 
-Clone the repository:
+First, clone the repository:
 ```
 git clone https://github.com/UZ-SLAMLab/ORB_SLAM3.git ORB_SLAM3
 ```
+**Android:**
+Clone the dependencies, execute:
+```
+cd android_build
```
Comment (guyrodoy1) [inline README.md:91]: The script should do the 'cd' command, assume calling the script from the root folder of this repo:
./android_build/clone_deps.h

That way we wont have to write wrapper scripts in order to run in in ci.

File: README.md [hunk 2]
```diff
@@ -81,11 +81,26 @@ We provide some examples to process input of a monocular, monocular-inertial, st
...
+chmod +x clone_deps.sh
+./clone_deps.sh
+```
+
+Then, to build ORB_SLAM3 and to copy the required files to DemoApp execute:
+```
+chmod +x build_android.sh
+./build_android.sh --app_path PATH_TO_DEMOAPP
```
Comment (guyrodoy1) [inline README.md:99]: The script should do the 'cd' command, assume calling the script from the root folder of this repo:
./android_build/build_android.h

That way we wont have to write wrapper scripts in order to run it in ci.

File: android_build/clone_deps.sh [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    exit 1
+}
+
+# Set up the trap to call the handle_error function
+trap 'handle_error' ERR
+
+# help menu
+function script_help {
+    echo "Usage: clone_deps.sh [parameters]"
+    echo -e "  -f<optional>\t\t\tOverwrite existing files and folders"
```
Comment (guyrodoy1) [inline android_build/clone_deps.sh:36]: Why not clean? Then you can delete before the clone

File: include/LocalMapping.h [hunk 4]
```diff
@@ -105,7 +105,7 @@ class LocalMapping
     bool mbNotBA2;
     bool mbBadImu;
 
-    bool mbWriteStats;
+    // bool mbWriteStats;
```
Comment (guyrodoy1) [inline include/LocalMapping.h:108]: delete

File: android_build/clone_deps.sh [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+# reset Thirdparty
+function delete_libs {
+  cd $BUILD_DIR/../Thirdparty
+  rm -r -f eigen-$EIGEN_VERSION Boost-for-Android OpenCV-android-sdk openssl
+}
+
+# Define a function to handle errors
+handle_error() {
+    echo "An error occurred. Exiting..."
+    read -p "Delete downloaded libs? (Y/N): " DELETE
```
Comment (guyrodoy1) [inline android_build/clone_deps.sh:23]: This is problematic for CI. you can an option of -y to your scripy (allways yes)

File: android_build/build_android.sh [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+trap 'handle_error' ERR
+
+# help menu
+function script_help {
+    echo "Usage: build_android.sh [parameters]"
+    echo -e "  --app_path\t\t\tAbsolute path to DemoApp root"
+    echo -e "  --ndk_path\t\t<optional> Absolute path to NDK (default is /home/${USER}/Android/Sdk/ndk/25.1.8937393)"
+    echo -e "  --abi\t\t<optional> android abi to build for (arm64-v8a, armeabi-v7a, x86, x86_64). default is arm64-v8a"
+    echo -e "  --platform\t\t<optional> android api level to build for (default is 30)"
+    echo -e "  --help\t\t\tsee this help"
```
Comment (guyrodoy1) [inline android_build/build_android.sh:32]: Add --clean

**Review by guyrodoy1 — CHANGES_REQUESTED** [review]

File: android_build/clone_deps.sh (hunk 5 above)
Comment (ron-lbb) [reply android_build/clone_deps.sh:23]: We can use yes command instead.

**Review by guyrodoy1 — APPROVED** [review]

**Pull request merged**